.github/workflows/versioning.yml

```diff
@@ -0,0 +1,170 @@
+# Workflow to verify versioning compatibility between main and patch versions
+name: Verify Versioning Compatibility
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version tag to compare against main (e.g., v1.2.0)'
+        required: true
+        default: 'v1.2.0'
+        type: string
+
+jobs:
+  verify-versioning:
+    runs-on:
+      - runs-on=${{ github.run_id }}
+      - runner=64cpu-linux-arm64
+      - extras=s3-cache
+    steps:
+      - name: Set up environment
+        uses: actions/checkout@v4
+
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: stable
+
+      - name: Cache Cargo dependencies
+        uses: Swatinem/rust-cache@v2
+
+      # Build and test from main
+      - name: Build main CLI
+        run: cargo install --force --path crates/cli
+
+      - name: Run setup from main
+        run: |
+          mkdir -p ./main-outputs
+          cargo openvm setup --output-dir ./main-outputs
+
+      - name: Build and keygen examples from main
+        run: |
+          mkdir -p ./main-outputs/examples
+          for example in examples/*/; do
+            if [ -f "$example/Cargo.toml" ]; then
+              example_name=$(basename "$example")
+              echo "Building and generating keys for example: $example_name"
+              cargo openvm build --manifest-path "$example/Cargo.toml"
+              cargo openvm keygen --manifest-path "$example/Cargo.toml" --output-dir "./main-outputs/examples/$example_name"
+            fi
+          done
+
+      - name: Build and keygen benchmarks from main
+        run: |
+          mkdir -p ./main-outputs/benchmarks
+          for benchmark in benchmarks/guest/*/; do
+            if [ -f "$benchmark/Cargo.toml" ]; then
+              benchmark_name=$(basename "$benchmark")
+              echo "Building and generating keys for benchmark: $benchmark_name"
+              cargo openvm build --manifest-path "$benchmark/Cargo.toml"
+              cargo openvm keygen --manifest-path "$benchmark/Cargo.toml" --output-dir "./main-outputs/benchmarks/$benchmark_name"
+            fi
+          done
+
+      # Checkout and test tagged version
+      - name: Checkout tagged version
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.version }}
```

Comment:
isn't `inputs` for workflow_call, you need `github.events.inputs` (check this) for workflow_dispatch?